packages/vote/vote_context.go

```diff
@@ -74,3 +78,29 @@ func (vc *Context) IsNew() bool {
 func (vc *Context) HadFirstRound() bool {
 	return vc.Rounds == 1
 }
+
+// HadFixedRound tells whether the vote context is in the last l2 rounds of fixed threshold
+func (vc *Context) HadFixedRound(coolingOffPeriod int, finalizationThreshold int, fixedEndingThreshold int) bool {
+	consecutiveRounds := finalizationThreshold - fixedEndingThreshold // l - l2
+	// check whether we have enough opinions to say whether this vote context is finalized.
```

Comment:
check whether we have enough opinions to say whether the random threshold for this vote context can be fixed.

Reply:
corrected